k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -5,52 +5,48 @@
 import java.net.SocketException;
 import java.util.ArrayList;
 import java.util.Collections;
-import java.util.Iterator;
-import java.util.LinkedList;
 import java.util.List;
 
 import android.content.Context;
 import android.os.PowerManager;
 
 import com.fsck.k9.mail.AuthenticationFailedException;
-import com.fsck.k9.mail.Flag;
 import com.fsck.k9.mail.K9MailLib;
-import com.fsck.k9.mail.Message;
 import com.fsck.k9.mail.MessagingException;
 import com.fsck.k9.mail.PushReceiver;
 import com.fsck.k9.mail.power.TracingPowerManager;
 import com.fsck.k9.mail.power.TracingPowerManager.TracingWakeLock;
 import com.fsck.k9.mail.store.RemoteStore;
 import timber.log.Timber;
 
+import static com.fsck.k9.mail.Folder.OPEN_MODE_RO;
 import static com.fsck.k9.mail.K9MailLib.PUSH_WAKE_LOCK_TIMEOUT;
 import static com.fsck.k9.mail.store.imap.ImapResponseParser.equalsIgnoreCase;
 
 
-class ImapFolderPusher extends ImapFolder {
+class ImapFolderPusher {
     private static final int IDLE_READ_TIMEOUT_INCREMENT = 5 * 60 * 1000;
     private static final int IDLE_FAILURE_COUNT_LIMIT = 10;
     private static final int MAX_DELAY_TIME = 5 * 60 * 1000; // 5 minutes
     private static final int NORMAL_DELAY_TIME = 5000;
 
-
+    private final ImapFolder folder;
```

Comment:
I'm not sure how to do that, since the folder / the folder name needs to saved in between calls to the constructor and the `start()` / `stop()` methods. Since each instance of `ImapFolderPusher` manages a single folder, I think it is appropriate to have it as an instance variable.